app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt

```diff
@@ -455,3 +463,126 @@ private fun UserReviewsSection(
+          contentPadding = PaddingValues(16.dp)) {
+            itemsIndexed(imagesUrls) { index, url ->
+              UserImageCard(
+                  url = url,
+                  onClick = {
+                    showDialogImage.value = url
+                    showDialogImageDestinationPath.value =
+                        showDialogImageDestinationPath.value.plus(it)
+                    showDialogParking.value =
+                        parkingViewModel.getParkingFromImagePath(imagesPaths[index])
+                    showDialogImagePath.value = imagesPaths[index]
+                  })
+            }
+          }
+    }
+  }
+
+  // Show dialog if an image is clicked
+  showDialogImage.value?.let { imageUrl ->
+    AlertDialog(
+        onDismissRequest = { showDialogImage.value = null },
+        title = { Text(stringResource(R.string.view_profile_screen_image_dialog_title)) },
+        text = {
+          Column {
+            Text(stringResource(R.string.view_profile_screen_image_dialog_description))
+            Spacer(modifier = Modifier.height(16.dp))
+            Image(
+                painter = rememberImagePainter(data = imageUrl),
+                contentDescription = null,
+                modifier = Modifier.fillMaxWidth().height(200.dp))
+          }
+        },
+        confirmButton = {
+          IconButton(
+              modifier = Modifier.padding(10.dp),
+              icon = Icons.Filled.LocalParking,
+              contentDescription = "Go To Parking",
+              testTag = "ParkingFromImageButton",
+              onClick = {
+                parkingViewModel.getParkingById(
+                    showDialogParking.value!!,
+                    { parkingForPath ->
+                      parkingViewModel.selectParking(parkingForPath)
+                      navigationActions.navigateTo(Screen.PARKING_DETAILS)
+                      Toast.makeText(context, strResToast, Toast.LENGTH_SHORT).show()
+                    },
+                    {})
+              })
+        },
+        dismissButton = {
+          IconButton(
+              modifier = Modifier.padding(10.dp),
+              icon =
+                  Icons.Filled
+                      .RestoreFromTrash, // or Icons.Filled.Paid or Icons.Filled.AttachMoney etc...
+              contentDescription = "Delete",
+              testTag = "DeleteImageButton",
+              onClick = {
+                userViewModel.removeImageFromUserImages(showDialogImagePath.value!!)
+                parkingViewModel.getParkingById(
+                    showDialogParking.value!!,
+                    { parkingForPath ->
+                      parkingViewModel.selectParking(parkingForPath)
+                      parkingViewModel.deleteImageFromParking(
+                          parkingViewModel.selectedParking.value?.uid!!,
+                          showDialogImagePath.value!!)
+                      navigationActions.navigateTo(Screen.VIEW_PROFILE)
```

Comment:
You mentioned that a reboot of the app is needed for the change to take effect. Do you think that some kind of "refresh" function in the view model could be done for this ? I think this might help with the reordering problem found by @sean-prz.

Reply:
I've addressed all of your comments except this one in my most recent commits. This one's been a long standing issue (not something this PR introduces) that I think warrants its own PR as this is an issue that I've been fighting a lot time now and haven't managed to find a proper fix for quite yet. Does that work for you?

Reply:
EDIT: I managed to fix this, partially. The weird "shuffling" around of images that Sean brought to my attention is fixed, but a reload of the app is still necessary to see the image updates take effect, for a reason still unknown by me for now. I will keep looking.

Reply:
I think that's as good as this PR will go ^^ Thank you for doing your best on this !

Reply:
I just had an idea for this. I'm pretty happy with the fact that the user deleting his Image from the profile screen reflects on the Parking, that's pretty huge for me since that the majority use case. Since it doesn't work on the Parking, though, I'm thinking of changing the DetailsScreen toast to "Image deleted, please reload the app to see change!". What do you think?